CovertInfiltration/Src/CovertInfiltration/Classes/XComGameState_ActivityChainSpawner.uc

```diff
@@ -311,21 +313,32 @@ static function SpawnCounterDarkEvents (XComGameState NewGameState)
 
 	// Step 1: spawn the chains
 
+	RegionRefs = GetContactedRegions();
+
 	foreach AlienHQ.ChosenDarkEvents(DarkEventRef)
```

Comment:
You should randomize the order so that the player can't predict which DEs they will be able to counter if there are less contacted regions than the DEs